modules/core/src/v2/coins/eth.ts

```diff
@@ -886,6 +893,102 @@ export class Eth extends BaseCoin {
+
+    if (typeof(sendAmount) !== 'string' || !sendAmount.match(/^-?[0-9]+$/)) {
+      throw Error('total amount to send must be in string form');
+    }
+
+    if (sendAddress !== common.Environments[this.bitgo.env].batcherContractAddress) {
+      throw new Error('batch transactions must be addressed to the Batcher')
+    }
+
+    const functionSignatureHash = batchData.slice(2, 10);
+    if (functionSignatureHash !== Eth.batcherFunctionSignature) {
+      throw new Error('batch data function signature hash does not match expected');
+    }
+
+    const [addresses, amounts] = optionalDeps.ethAbi.rawDecode(Eth.batchMethodTypes, new Buffer(batchData.slice(10), 'hex'));
+    if (addresses.length !== amounts.length) {
+      throw new Error('recipients and amounts arrays must be the same size');
+    }
+
+    const batchRecipients = [];
+    let totalAmount = new BigNumber(0);
+    for (let i = 0; i < addresses.length; i++) {
+      const address = addresses[i].toString('hex');
+      if (!this.isValidAddress(address)) {
+        throw new Error('invalid address: ' + address);
+      }
+
+      const amount = amounts[i].toString();
+      if (typeof(amount) !== 'string' || !amount.match(/^-?[0-9]+$/)) {
+        throw new Error('invalid amount: ' + amounts[i].toString());
+      }
+      totalAmount = totalAmount.plus(amount);
+
+      batchRecipients.push({
+        address,
+        amount,
+      });
+    }
+
+    const sortedBatchRecipients = _(batchRecipients).sortBy(['address', 'amount'])
+    .map(({ address, amount }) => ({
+      address: optionalDeps.ethUtil.addHexPrefix(address),
+      amount: new BigNumber(amount).toFixed(),
+    }))
+    .value();
+
+    const sortedOriginalRecipients = _(originalRecipients).sortBy(['address', 'amount'])
+    .map(({ address, amount }) => ({
+      address: optionalDeps.ethUtil.addHexPrefix(address),
+      amount: new BigNumber(amount).toFixed(),
+    }))
+    .value();
+
+    if (!_.isEqual(sortedBatchRecipients, sortedOriginalRecipients)) {
+      throw new Error('batch recipients in data does not match original recipients')
+    }
+
+    if (!totalAmount.eq(sendAmount)) {
+      throw new Error(`Batcher send amount ${sendAmount} is not equal to sum of individual transfers ${totalAmount.toFixed()}`);
+    }
+  }
+
+  /**
+   * Validates that the prebuild data returned by platform is valid
+   * @param params
+   * @param callback
+   * @throws Error if the prebuild is invalid
+   */
+  verifyTransaction(params: any, callback?: NodeCallback<any>): Bluebird<any> {
+    return co(function *() {
+      const { txParams, txPrebuild } = params;
+      if (txPrebuild.recipients.length !== 1) {
+        throw new Error('prebuild should always have recipients array of length 1');
+      }
+      if (txParams.recipients.length === 1 && !_.isEqual(txPrebuild.recipients, txParams.recipients)) {
+        throw new Error('recipients returned by platform are not equal to recipients in original tx request');
```

Comment:
Let's call it BitGo instead of platform, customers don't know what platform is.

Reply:
Changed.